pkg/k8shandler/fluentd.go

```diff
@@ -415,9 +416,42 @@ func newFluentdPodSpec(cluster *logging.ClusterLogging, elasticsearchAppName str
 	// Shorten the termination grace period from the default 30 sec to 10 sec.
 	fluentdPodSpec.TerminationGracePeriodSeconds = utils.GetInt64(10)
 
+	if !pipelineSpec.DisableDefaultForwarding {
+		fluentdPodSpec.InitContainers = []v1.Container{
+			newFluentdInitContainer(cluster),
+		}
+	}
+
 	return fluentdPodSpec
 }
 
+func newFluentdInitContainer(cluster *logging.ClusterLogging) v1.Container {
+	collectionSpec := logging.CollectionSpec{}
+	resources := collectionSpec.Logs.FluentdSpec.Resources
+	if resources == nil {
+		resources = &v1.ResourceRequirements{
+			Limits: v1.ResourceList{v1.ResourceMemory: defaultFluentdMemory},
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: defaultFluentdMemory,
+				v1.ResourceCPU:    defaultFluentdCpuRequest,
+			},
+		}
+	}
+	initContainer := NewContainer("fluentd-init", "fluentd", v1.PullIfNotPresent, *resources)
```

Comment:
Is there any reason not to reuse `fluentName`

Reply:
i wanted to be explicit that this was an init container for cases when viewing logs or if we needed to exec/rsh into it